sdk/storage/azure-storage-blob/azure/storage/blob/_shared/encryption.py

```diff
@@ -115,23 +154,32 @@ def __init__(self, content_encryption_IV, encryption_agent, wrapped_content_key,
             A dict containing metadata related to the key wrapping.
         '''
 
-        _validate_not_none('content_encryption_IV', content_encryption_IV)
         _validate_not_none('encryption_agent', encryption_agent)
         _validate_not_none('wrapped_content_key', wrapped_content_key)
 
+        # Validate we have the right info for the specified algorithm
+        if encryption_agent.encryption_algorithm == _EncryptionAlgorithm.AES_CBC_256:
+            _validate_not_none('content_encryption_IV', content_encryption_IV)
+        elif encryption_agent.encryption_algorithm == _EncryptionAlgorithm.AES_GCM_256:
+            _validate_not_none('encrypted_region_info', encrypted_region_info)
+        else:
+            raise ValueError("Invalid encryption algorithm.")
```

Comment:
This is probably fine, but just a thought -- maybe a more specific ValueError? Since technically they could have specified a valid encryption algorithm but just didn't provide the matching required parameter (but the ValueError suggests otherwise)

Reply:
This `else` actually DOES mean the actual encryption algorithm was invalid, not that we had missing parameters for it, so I think this error is appropriate. Also, this is an internal method, and I don't think a customer would ever actually see this, more here to be extra cautious, I guess. A customer would have to modify their metadata in such a way to get this.